### PR TITLE
craftable stethoscope

### PR DESCRIPTION
**Pull request by misterprimus**
#### Summary

```SUMMARY: Content "craftable stethoscope"```
#### Purpose of change

To fix some things in the game that don't make sense from a realism/balance perspective.
#### Describe the solution

1. Allow crafting of stethoscopes. These are a pain to find, but you can't ever craft them, even though online tutorials show even a child can do it:
https://www.thoughtco.com/make-your-own-stethoscope-2086570
While not exactly something a doctor would ever show to a patient, it seems as though it should work for simple listening as in safe cracking.
#### Testing

Tested fix to see if behavior resulted in expected behavior and corrected unexpected behavior accordingly.
#### Additional context

N/A

### CODE REVIEW

File: data/json/recipes/recipe_others.json [hunk 1]
```diff
@@ -4666,5 +4666,17 @@
...
+  },
+  {
+    "type": "recipe",
+    "result": "stethoscope",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_TOOLS",
+    "skill_used": "fabrication",
+    "difficulty": 1,
+    "time": "15 m",
+    "autolearn": true,
```
Comment (DaviBones) [inline data/json/recipes/recipe_others.json]: While a child could *make* this recipe, I doubt that even many *adults* would think of it on their own. I'd like to see a book this recipe is found in, rather than having it auto-learned.

File: data/json/recipes/recipe_others.json [hunk 2]
```diff
@@ -4666,5 +4666,17 @@
...
+    "qualities": [ { "id": "CUT", "level": 1 } ],
+    "components": [ [ [ "duct_tape", 30 ] ], [ [ "bottle_plastic_small", 2 ] ], [ [ "hose", 2 ] ] ]
```
Comment (DaviBones) [inline data/json/recipes/recipe_others.json]: What about the funnel? We have several types in-game, no reason to not have them as alternatives in the recipe.

Also, why 2 rubber hoses rather than one?

**Comment by Fris0uman** [comment]
Adding balloons to the game would be nice, alternatively we have condom already.

File: data/json/recipes/recipe_others.json (hunk 1 above)
Comment (misterprimus) [reply data/json/recipes/recipe_others.json]: I'll see if I can find a reasonable book(s) to contain this recipe.

File: data/json/recipes/recipe_others.json (hunk 2 above)
Comment (misterprimus) [reply data/json/recipes/recipe_others.json]: > 
> 
> What about the funnel? We have several types in-game, no reason to not have them as alternatives in the recipe.
> 
> Also, why 2 rubber hoses rather than one?

Since there's no "balloon" item in the game, I figured a second hose could be cut and pulled over the other end to provide the membrane. Didn't even think of the condom. Definitely makes more sense. Will use that instead.

I think I'll add the funnel as an alternative to the small bottles.

**Comment by misterprimus** [comment]
> 
> 
> Adding balloons to the game would be nice, alternatively we have condoms already.

Yes I will take the condom approach.

**Comment by Rivet-the-Zombie** [comment]
This makeshift recipe probably shouldn't be producing the same slick little stethoscope that you can find in doctor's offices, but rather a bigger, heavier makeshift version.

**Comment by misterprimus** [comment]
OK I think everything should be good now.

**Comment by misterprimus** [comment]
ok all done

**Comment by misterprimus** [comment]
Done, but suddenly getting 
image

File: data/json/items/containers.json [hunk 3]
```diff
@@ -1140,6 +1140,26 @@
...
+    "price": 0,
+    "to_hit": -5,
+    "rigid": false,
+    "material": "plastic",
+    "symbol": ")",
+    "color": "white",
+    "contains": "3750 ml",
+    "watertight": true,
+    "seals": true,
+    "properties": [ [ "burst_when_filled", "75" ] ]
```
Comment (DaviBones) [inline data/json/items/containers.json]: A lot of these entries could be removed if you add `"copy-from": "condom"`.
Or perhaps go the other direction and have condom inherit from balloon.

**Comment by mark7** [comment]
I have to say that I'm not enthusiastic about this pull request from a gameplay standpoint.

This change will make it substantially-easier to get into safes.  One will need fabrication(1), one of several common books, 1 hose from a smashed fridge, 2 commonly-dropped plastic bottles, and 30 duct tape...and safes become openable.  This avoids some of the reason for doing dangerous raids into doctor's offices in cities to get a stethoscope.

My thinking is that the gates that the map makers can place on accessing items are already pretty low.  The general aim seems to be to try to extend the game before one gets to "uber-character", and gating access to safes seems to be an important part of this.

I don't know whether it's realistic or not to crack a safe with the items listed here.  But I do wonder if it's a good idea to permit it from a gameplay standpoint.

If one makes a realism argument, then it might be fair to ask whether an average Joe who comes upon the above-mentioned items could reasonably crack a safe with them and the aid of a first-aid handbook.

I raised a similar issue with issue #31817, that makeshift lockpicks with their current recipe trivialized the bar to get through locked doors on "vaults" like gun stores.  @kevingranade agreed, but there were objections from others that crafting a makeshift lockpick was not hard, but that being able to *use* one should be harder...and I think that they've a certain point.

Much as I dislike adding load to the patch creator, would it be reasonable, given that the bar is being lowered to create the tools to get into safes and that this is effectively lowering the bar to get into a safe, to also raise the bar to effectively *use* those tools?  Bump up the skill requirements there?

**Comment by Zireael07** [comment]
Does the stethoscope item have any uses apart from cracking safes?

**Comment by misterprimus** [comment]
> 
> 
> I have to say that I'm not enthusiastic about this pull request from a gameplay standpoint.
> 
> This change will make it substantially-easier to get into safes. One will need fabrication(1), one of several common books, 1 hose from a smashed fridge, 2 commonly-dropped plastic bottles, a condom or balloon, and 30 duct tape/medical tape...and safes become openable. This avoids some of the reason for doing dangerous raids into doctor's offices in cities to get a stethoscope.
> 
> My thinking is that the gates that the map makers can place on accessing items are already pretty low. The general aim seems to be to try to extend the game before one gets to "uber-character", and gating access to safes seems to be an important part of this.
> 
> I don't know whether it's realistic or not to crack a safe with the items listed here. But I do wonder if it's a good idea to permit it from a gameplay standpoint.
> 
> If one makes a realism argument, then it might be fair to ask whether an average Joe who comes upon the above-mentioned items could reasonably crack a safe with them and the aid of a first-aid handbook.
> 
> I raised a similar issue with issue #31817, that makeshift lockpicks with their current recipe trivialized the bar to get through locked doors on "vaults" like gun stores. @kevingranade agreed, but there were objections from others that crafting a makeshift lockpick was not hard, but that being able to _use_ one should be harder...and I think that they've a certain point.
> 
> Much as I dislike adding load to the patch creator, would it be reasonable, given that the bar is being lowered to create the tools to get into safes and that this is effectively lowering the bar to get into a safe, to also raise the bar to effectively _use_ those tools? Bump up the skill requirements there?
> 
> What I'm getting at is that I'm not sure that having safes be hard to open is a bug, at least from a gameplay standpoint -- I'd call it a feature. It permits separating "later game" items from "immediately-available" items.

A few points here:

1. Basic contact vibration rerouting devices like a basic stethoscope IS quite simple to make and these are sold quite inexpensively for car mechanics to search for air leaks (effectiveness is debatable where that is concerned).
2. Safes in bank vaults appear to have been nerfed? I remember months ago a raid could easily result in multiple powered armor suits/helmets, whereas now the best I can hope for is some advanced books/manuals. Rest has been replaced with money and jewelry, which I suppose makes sense.
3. You can't just get into a safe with a stethoscope as it is. You must still clear out the mobs (banks are often in city centers in big cities), beat the lethal secubots, crack the computer (needs very high computers skill) or smash the wall (requires jackhammer/pickaxe), etc... IRL a safe is there to slow down an intruder, not stop him. There's no solution in a typical commercial safe against an electric saw or similar item and unlike in the game this likely would not damage the contents.
4. I would personally be able to craft a stethoscope fairly easily, but I'd have no clue how to crack a safe, especially a modern commercial one like that of a bank vault (is it really as simple as listening to the clicks it makes when you turn the dial?).

Perhaps the solution is to make safe-cracking something that can only be possible above a certain mechanics level or perhaps have a high chance of failure (with no penalty besides wasted time) at lower mechanics levels? But if this is done I think it should also be made possible (if time consuming) to SAFELY crack open a safe with a power drill or angle grinder.

**Comment by mark7** [comment]
> Does the stethoscope item have any uses apart from cracking safes?

Not in the base game -- there are no references to the item id "stethoscope", and the only relevant flag on the item is SAFECRACK, which is only used for safecracking.  I suppose that someone may have a mod that I'm unaware of that does something else with it.

**Comment by mark7** [comment]
> Safes in bank vaults appear to have been nerfed?  I remember months ago a raid could easily result in multiple powered armor suits/helmets, whereas now the best I can hope for is some advanced books/manuals

That looks like a34a096b1d549d6a89890959fcabb7c15141cb99, yeah.

> You must still clear out the mobs (banks are often in city centers in big cities), beat the lethal secubots, crack the computer (needs very high computers skill) or smash the wall (requires jackhammer/pickaxe), etc

I mean, sure, but if the safes aren't meant to be a substantial gate to the player, then what's the purpose of having them, in gameplay terms?  I mean, I normally have the stuff the recipe requires even without shooting for it.  I mean, as you point out, there's already a surrounding bank vault -- why not just have the stuff in the vault?

Like, there are a number of points where safes are used in the game, and I believe that they're generally-intended to gate "better" items from "not as good" items.  I currently see f_safe_l referenced in bank.json, basement_meth.json, jewel_store.json, mall.json, mansion.json, office_cubical.json, pawn_shop.json, police_station.json, post_office.json, prison_1.json, river_shipwreck.json, s_hunting.json, veterinarian.json, house_w_palette.json, mall_palette.json, microlab.json, office.json, robocachq.json; Aftershock's map_palettes.json; Fuji_Structures' s_lightindustry_10.json; More_Locations' bandit tower and tpalettes; Salvaged_Robots' tpalettes.json.

> Perhaps the solution is to make safe-cracking something that can only be possible above a certain mechanics level or perhaps have a high chance of failure (with no penalty besides wasted time) at lower mechanics levels?

Well, IME, time at a safe isn't *too* much of a constraint for me.  Most places that have vaults are reasonably comfortable for the character to be spending time in, food isn't usually presently too-severe of a constraint.

(Also, in practice, I'd assume that it'd make sense from a quality-of-life standpoint to just subsume failures into it taking longer to crack a safe.)

But, again, that seems to lower the bar quite a bit.

I mean, I guess the real question is -- what's the main purpose of the PR?  Is it primarily to address realism, or is it to alter gameplay, to lower the bar to get into safes?

If it's to address realism, for all I know, it's viable to make a stethoscope from these parts...but then, I have to say that I doubt that it'd be enough to get me in-real-life into a safe.  And it seems like if the game goes in to address the former realism issue without also addressing the latter realism element, then it impacts gameplay.

From a realism standpoint, I imagine that I could probably eventually get into most safes (as you point out) with an angle grinder, agreed -- and maybe adding the ability to get into safes with an angle grinder is a fair change -- and from a gameplay standpoint, as long as they show up reasonably-infrequently, they don't kill the bar.

If it's to address gameplay, to lower the bar to get into safes -- you said that it was "a pain to find" stethoscopes, so I'm under the belief that that's at least a part goal -- then that's where I start to worry about the resulting bar being too low.

I mean, from a gameplay standpoint, I'd say that it's okay for, on a given run, some characters -- like the Burglar profession -- being able to get into safes early, and others not being able to do.  Part of the appeal of roguelike games is in making each game different by throwing up a meaningfully-different set of constraints at the player each run.  If I know that I have a pretty reliable way to get into safes early in each run, then "character can get into safes" stops being one of those inputs, and the gameplay becomes pretty formulaic in each game.

**Comment by anothersimulacrum** [comment]
The triviality of getting into safes is the problem, not the ease of crafting the stethoscope.
It's also out of scope of this PR. I'd say make a separate issue or PR to resolve that.

**Comment by misterprimus** [comment]
> > Safes in bank vaults appear to have been nerfed?  I remember months ago a raid could easily result in multiple powered armor suits/helmets, whereas now the best I can hope for is some advanced books/manuals
> 
> That looks like [a34a096](https://github.com/CleverRaven/Cataclysm-DDA/commit/a34a096b1d549d6a89890959fcabb7c15141cb99), yeah.
> 
> > You must still clear out the mobs (banks are often in city centers in big cities), beat the lethal secubots, crack the computer (needs very high computers skill) or smash the wall (requires jackhammer/pickaxe), etc
> 
> I mean, sure, but if the safes aren't meant to be a substantial gate to the player, then what's the purpose of having them, in gameplay terms? I mean, I normally have the stuff the recipe requires even without shooting for it. I mean, as you point out, there's already a surrounding bank vault -- why not just have the stuff in the vault?
> 
> Like, there are a number of points where safes are used in the game, and I believe that they're generally-intended to gate "better" items from "not as good" items. I currently see f_safe_l referenced in bank.json, basement_meth.json, jewel_store.json, mall.json, mansion.json, office_cubical.json, pawn_shop.json, police_station.json, post_office.json, prison_1.json, river_shipwreck.json, s_hunting.json, veterinarian.json, house_w_palette.json, mall_palette.json, microlab.json, office.json, robocachq.json; Aftershock's map_palettes.json; Fuji_Structures' s_lightindustry_10.json; More_Locations' bandit tower and tpalettes; Salvaged_Robots' tpalettes.json.
> 
> > Perhaps the solution is to make safe-cracking something that can only be possible above a certain mechanics level or perhaps have a high chance of failure (with no penalty besides wasted time) at lower mechanics levels?
> 
> Well, IME, time at a safe isn't _too_ much of a constraint for me. Most places that have vaults are reasonably comfortable for the character to be spending time in, food isn't usually presently too-severe of a constraint.
> 
> (Also, in practice, I'd assume that it'd make sense from a quality-of-life standpoint to just subsume failures into it taking longer to crack a safe.)
> 
> But, again, that seems to lower the bar quite a bit.
> 
> I mean, I guess the real question is -- what's the main purpose of the PR? Is it primarily to address realism, or is it to alter gameplay, to lower the bar to get into safes?
> 
> If it's to address realism, for all I know, it's viable to make a stethoscope from these parts...but then, I have to say that I doubt that it'd be enough to get me in-real-life into a safe. And it seems like if the game goes in to address the former realism issue without also addressing the latter realism element, then it impacts gameplay.
> 
> From a realism standpoint, I imagine that I could probably eventually get into most safes (as you point out) with an angle grinder, agreed -- and maybe adding the ability to get into safes with an angle grinder is a fair change -- and from a gameplay standpoint, as long as they show up reasonably-infrequently, they don't kill the bar.
> 
> If it's to address gameplay, to lower the bar to get into safes -- you said that it was "a pain to find" stethoscopes, so I'm under the belief that that's at least a part goal -- then that's where I start to worry about the resulting bar being too low.
> 
> I mean, from a gameplay standpoint, I'd say that it's okay for, on a given run, some characters -- like the Burglar profession -- being able to get into safes early, and others not being able to do. Part of the appeal of roguelike games is in making each run different by throwing up a meaningfully-different set of constraints at the player each run. If I know that I have a pretty reliable way to get into safes early in each run, then "character can get into safes" stops being one of those inputs, and the gameplay becomes pretty formulaic in each game.
> 
> Right now, obtaining the ability to get into a safe means, for most characters, either being lucky, or having to scout around -- possibly even multiple towns -- until they can find a doctor's office, and then manage to get into it and search it. That's a significant project. It forces characters to move around and forces them to undertake sub-projects along the way. Sure, I mean, it takes time...but is "it's a substantial undertaking to get the ability to loot safes" a problem? I mean, it seems to me that the game generally benefits from having those large projects and requiring that travel and exploration and benefits from that degree of unpredictability.

I see the "safes" as internal safe deposit boxes, with the double metal doors acting as the real "vault door" that must be breached. This makes far more sense now that the vault loot is a lot more sensible. Frankly, the secubot loot might be the biggest prize to be had in some banks. I thnk there already is loot spawning on the bank floor near the safes.

As for what's in safes, that's anybody's guess. Right now the loot is hardly anything to write home about, as it should be since most folks' possessions are quite mundane - cheap jewelry and money. Maybe some gold. Nothing earthshattering in terms of gameplay.

As for why I wanted to do this PR, in addition to realism, the primary purpose is probably to get rid of the seemingly arbitrary and incredible luck-based approach to letting people get into safes. Basically, it just wasn't making sense to me that I needed to search for some seemingly magical stethoscope like it is The Chalice when:
1. IRL I could make it myself, find it in ANY doctor's office,  etc...
2. A safe could easily be beaten by an angle grinder, an utterly mundane and extremely common (not UNcommon as you suggest) item, sold at Harbor Freight for $15!
3. IRL the most difficult part of breaking into a safe: police (secubots, mobs) and the vault door itself, which is difficult to break into by simple virtue of how big and thick it is.

Basically it was absurd that I did all this hard work of slaughtering a small army and smashing through a thick carbon steel door only to be undone by a safe deposit box I could IRL break open with a claw hammer and a chisel.

Frankly, I don't see craftable stethoscopes as a negative gameplay balance issue anymore since the safe contents were rightly nerfed. At best it's a realism/"feel" issue of making safe cracking by stethoscope a mechanics skill and at the same time allowing one to do it with an angle grinder. Nothing so urgent that it needs to be lumped into this PR.
